### PR TITLE
Move to CircleCI

### LINKED ISSUE #106: Migrate to CircleCI
We are moving everything to CircleCI, so we should move this project as well.

See https://github.com/readthedocs/sphinx-notfound-page/pull/130

Be sure to keep the tests on _all the Sphinx minor versions_.

### LINKED ISSUE #137: 0.6b1: pytest is failing
Just normal build, install and test cycle used on building package from non-root account:
- "setup.py build"
- "setup.py install --root </install/prefix>"
- "pytest with PYTHONPATH pointing to setearch and sitelib inside </install/prefix>
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-sphinx-hoverxref-0.6b1-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-sphinx-hoverxref-0.6b1-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ PYTHONDONTWRITEBYTECODE=1
+ /usr/bin/pytest -ra
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.11, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /home/tkloczko/rpmbuild/BUILD/sphinx-hoverxref-0.6b1, configfile: pytest.ini
plugins: forked-1.3.0, shutil-1.7.0, virtualenv-1.7.0, expect-1.1.0, httpbin-1.0.0, flake8-1.0.7, timeout-1.4.2, betamax-0.8.1, freezegun-0.4.2, case-1.5.3, isort-1.3.0, aspectlib-1.5.2, asyncio-0.15.1, toolbox-0.5, xprocess-0.17.1, aiohttp-0.3.0, checkdocs-2.7.0, mock-3.6.1, rerunfailures-9.1.1, requests-mock-1.9.3, cov-2.12.1, pyfakefs-4.5.0, cases-3.6.1, flaky-3.7.0, hypothesis-6.14.0, benchmark-3.4.1, xdist-2.3.0, pylama-7.7.1, datadir-1.3.1, regressions-2.2.0, Faker-8.9.0
collected 15 items

. .                                                                                                                                                                  [  6%]
tests/test_htmltag.py ........FFF                                                                                                                                    [ 80%]
tests/test_internals.py ...                                                                                                                                          [100%]

================================================================================= FAILURES =================================================================================
_____________________________________________________________________ test_intersphinx_default_configs _____________________________________________________________________

app = <SphinxTestApp buildername='html'>, status = <_io.StringIO object at 0x7f4e0c515790>, warning = <_io.StringIO object at 0x7f4e0c5158b0>

    @pytest.mark.sphinx(
        srcdir=intersphinxsrc,
    )
    def test_intersphinx_default_configs(app, status, warning):
        app.build()
        path = app.outdir / 'index.html'
        assert path.exists() is True
        content = open(path).read()

        chunks = [
            '<a class="reference external" href="https://docs.python.org/3/tutorial/index.html#tutorial-index" title="(in Python v3.9)"><span class="xref std std-ref">This a :ref: to The Python Tutorial using intersphinx</span></a>',
            '<a class="reference external" href="https://docs.python.org/3/library/datetime.html#datetime-datetime" title="(in Python v3.9)"><span class="xref std std-ref">This a :ref: to datetime.datetime Python’s function using intersphinx</span></a>',
            '<a class="reference external" href="https://docs.readthedocs.io/en/stable/config-file/v2.html#python" title="(in Read the Docs v5.17.0)"><span class="xref std std-ref">This a :ref: to Config File v2 Read the Docs’ page using intersphinx</span></a>',
            '<a class="reference external" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.9)"><code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code></a>',
            '<a class="reference internal" href="#hoverxref.extension.setup" title="hoverxref.extension.setup"><code class="xref py py-func docutils literal notranslate"><span class="pre">hoverxref.extension.setup()</span></code></a>',
        ]

        if sphinx.version_info >= (4, 0):
            chunks.extend([
                '<dt class="sig sig-object py" id="hoverxref.extension.setup">',
            ])
        else:
            chunks.extend([
                '<dt id="hoverxref.extension.setup">',
            ])


        for chunk in chunks:
>           assert chunk in content
E           assert '<a class="reference external" href="https://docs.readthedocs.io/en/stable/config-file/v2.html#python" title="(in Read....17.0)"><span class="xref std std-ref">This a :ref: to Config File v2 Read the Docs’ page using intersphinx</span></a>' in '\n<!DOCTYPE html>\n\n<html>\n  <head>\n    <meta charset="utf-8" />\n    <meta name="viewport" content="width=device-...ref="_sources/index.rst.txt"\n          rel="nofollow">Page source</a>\n    </div>\n\n    \n\n    \n  </body>\n</html>'

tests/test_htmltag.py:236: AssertionError
------------------------------------------------------------------------- Captured stdout teardown -------------------------------------------------------------------------
# testroot: root
# builder: html
# srcdir: /home/tkloczko/rpmbuild/BUILD/sphinx-hoverxref-0.6b1/tests/examples/intersphinx
# outdir: /home/tkloczko/rpmbuild/BUILD/sphinx-hoverxref-0.6b1/tests/examples/intersphinx/_build/html
# status:
Running Sphinx v4.0.2
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from https://docs.readthedocs.io/en/stable/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 1 source files that are out of date
updating environment: [new config] 1 added, 0 changed, 0 removed
reading sources... [100%] index
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] index
generating indices... genindex done
writing additional pages... search done
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in tests/examples/intersphinx/_build/html.

# warning:

_____________________________________________________________________ test_intersphinx_python_mapping ______________________________________________________________________

app = <SphinxTestApp buildername='html'>, status = <_io.StringIO object at 0x7f4deacbed30>, warning = <_io.StringIO object at 0x7f4deacbeb80>

    @pytest.mark.sphinx(
        srcdir=intersphinxsrc,
        confoverrides={
            'hoverxref_intersphinx': [
                'python',
            ],
        },
    )
    def test_intersphinx_python_mapping(app, status, warning):
        app.build()
        path = app.outdir / 'index.html'
        assert path.exists() is True
        content = open(path).read()

        chunks = [
            # Python's links do have hoverxref enabled
            '<a class="hoverxref tooltip reference external" data-url="https://docs.python.org/3/tutorial/index.html#tutorial-index" href="https://docs.python.org/3/tutorial/index.html#tutorial-index" title="(in Python v3.9)"><span class="xref std std-ref">This a :ref: to The Python Tutorial using intersphinx</span></a>',
            '<a class="hoverxref tooltip reference external" data-url="https://docs.python.org/3/library/datetime.html#datetime-datetime" href="https://docs.python.org/3/library/datetime.html#datetime-datetime" title="(in Python v3.9)"><span class="xref std std-ref">This a :ref: to datetime.datetime Python’s function using intersphinx</span></a>',
            '<a class="hoverxref tooltip reference external" data-url="https://docs.python.org/3/library/functions.html#float" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.9)"><code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code></a>',

            # Read the Docs' link does not have hoverxref enabled
            '<a class="reference external" href="https://docs.readthedocs.io/en/stable/config-file/v2.html#python" title="(in Read the Docs v5.17.0)"><span class="xref std std-ref">This a :ref: to Config File v2 Read the Docs’ page using intersphinx</span></a>',

            # Python's domain does not have hoverxref enabled
            '<a class="reference internal" href="#hoverxref.extension.setup" title="hoverxref.extension.setup"><code class="xref py py-func docutils literal notranslate"><span class="pre">hoverxref.extension.setup()</span></code></a>',
        ]

        for chunk in chunks:
>           assert chunk in content
E           assert '<a class="reference external" href="https://docs.readthedocs.io/en/stable/config-file/v2.html#python" title="(in Read....17.0)"><span class="xref std std-ref">This a :ref: to Config File v2 Read the Docs’ page using intersphinx</span></a>' in '\n<!DOCTYPE html>\n\n<html>\n  <head>\n    <meta charset="utf-8" />\n    <meta name="viewport" content="width=device-...ref="_sources/index.rst.txt"\n          rel="nofollow">Page source</a>\n    </div>\n\n    \n\n    \n  </body>\n</html>'

tests/test_htmltag.py:267: AssertionError
------------------------------------------------------------------------- Captured stdout teardown -------------------------------------------------------------------------
# testroot: root
# builder: html
# srcdir: /home/tkloczko/rpmbuild/BUILD/sphinx-hoverxref-0.6b1/tests/examples/intersphinx
# outdir: /home/tkloczko/rpmbuild/BUILD/sphinx-hoverxref-0.6b1/tests/examples/intersphinx/_build/html
# status:
Running Sphinx v4.0.2
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from https://docs.readthedocs.io/en/stable/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 1 source files that are out of date
updating environment: [new config] 1 added, 0 changed, 0 removed
reading sources... [100%] index
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
_hoverxref attributes: {'class': 'reference external', 'href': 'https://docs.python.org/3/tutorial/index.html#tutorial-index', 'title': '(in Python v3.9)', 'data-url': 'https://docs.python.org/3/tutorial/index.html#tutorial-index'}
_hoverxref attributes: {'class': 'reference external', 'href': 'https://docs.python.org/3/library/datetime.html#datetime-datetime', 'title': '(in Python v3.9)', 'data-url': 'https://docs.python.org/3/library/datetime.html#datetime-datetime'}
_hoverxref attributes: {'class': 'reference external', 'href': 'https://docs.python.org/3/library/functions.html#float', 'title': '(in Python v3.9)', 'data-url': 'https://docs.python.org/3/library/functions.html#float'}

generating indices... genindex done
writing additional pages... search done
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in tests/examples/intersphinx/_build/html.

# warning:

______________________________________________________________________ test_intersphinx_all_mappings _______________________________________________________________________

app = <SphinxTestApp buildername='html'>, status = <_io.StringIO object at 0x7f4dea9d6d30>, warning = <_io.StringIO object at 0x7f4dea9d60d0>

    @pytest.mark.sphinx(
        srcdir=intersphinxsrc,
        confoverrides={
            'hoverxref_intersphinx': [
                'readthedocs',
                'python',
            ],
            'hoverxref_intersphinx_types': {
                'readthedocs': 'modal',
                'python': {
                    'class': 'modal',
                }
            },
            'hoverxref_domains': ['py'],
        },
    )
    def test_intersphinx_all_mappings(app, status, warning):
        app.build()
        path = app.outdir / 'index.html'
        assert path.exists() is True
        content = open(path).read()

        chunks = [
            # Python's links do have hoverxref enabled
            '<a class="hoverxref tooltip reference external" data-url="https://docs.python.org/3/tutorial/index.html#tutorial-index" href="https://docs.python.org/3/tutorial/index.html#tutorial-index" title="(in Python v3.9)"><span class="xref std std-ref">This a :ref: to The Python Tutorial using intersphinx</span></a>',
            '<a class="hoverxref tooltip reference external" data-url="https://docs.python.org/3/library/datetime.html#datetime-datetime" href="https://docs.python.org/3/library/datetime.html#datetime-datetime" title="(in Python v3.9)"><span class="xref std std-ref">This a :ref: to datetime.datetime Python’s function using intersphinx</span></a>',
            '<a class="hoverxref modal reference external" data-url="https://docs.python.org/3/library/functions.html#float" href="https://docs.python.org/3/library/functions.html#float" title="(in Python v3.9)"><code class="xref py py-class docutils literal notranslate"><span class="pre">float</span></code></a>',

            # Read the Docs' link does have hoverxref enabled
            '<a class="hoverxref modal reference external" data-url="https://docs.readthedocs.io/en/stable/config-file/v2.html#python" href="https://docs.readthedocs.io/en/stable/config-file/v2.html#python" title="(in Read the Docs v5.17.0)"><span class="xref std std-ref">This a :ref: to Config File v2 Read the Docs’ page using intersphinx</span></a>',

            # Python domain's link does have hoverxref enabled
            '<a class="hoverxref tooltip reference internal" data-doc="index" data-docpath="/index.html" data-project="myproject" data-section="hoverxref.extension.setup" data-version="myversion" href="#hoverxref.extension.setup" title="hoverxref.extension.setup"><code class="xref py py-func docutils literal notranslate"><span class="pre">hoverxref.extension.setup()</span></code></a>',
        ]

        for chunk in chunks:
>           assert chunk in content
E           assert '<a class="hoverxref modal reference external" data-url="https://docs.readthedocs.io/en/stable/config-file/v2.html#pyt....17.0)"><span class="xref std std-ref">This a :ref: to Config File v2 Read the Docs’ page using intersphinx</span></a>' in '\n<!DOCTYPE html>\n\n<html>\n  <head>\n    <meta charset="utf-8" />\n    <meta name="viewport" content="width=device-...ref="_sources/index.rst.txt"\n          rel="nofollow">Page source</a>\n    </div>\n\n    \n\n    \n  </body>\n</html>'

tests/test_htmltag.py:306: AssertionError
------------------------------------------------------------------------- Captured stdout teardown -------------------------------------------------------------------------
# testroot: root
# builder: html
# srcdir: /home/tkloczko/rpmbuild/BUILD/sphinx-hoverxref-0.6b1/tests/examples/intersphinx
# outdir: /home/tkloczko/rpmbuild/BUILD/sphinx-hoverxref-0.6b1/tests/examples/intersphinx/_build/html
# status:
Running Sphinx v4.0.2
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from https://docs.readthedocs.io/en/stable/objects.inv...
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 1 source files that are out of date
updating environment: [new config] 1 added, 0 changed, 0 removed
reading sources... [100%] index
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
Using default style (tooltip) for unknown typ (func). Define it in hoverxref_role_types.
_hoverxref attributes: {'class': 'reference external', 'href': 'https://docs.python.org/3/tutorial/index.html#tutorial-index', 'title': '(in Python v3.9)', 'data-url': 'https://docs.python.org/3/tutorial/index.html#tutorial-index'}
_hoverxref attributes: {'class': 'reference external', 'href': 'https://docs.python.org/3/library/datetime.html#datetime-datetime', 'title': '(in Python v3.9)', 'data-url': 'https://docs.python.org/3/library/datetime.html#datetime-datetime'}
_hoverxref attributes: {'class': 'reference external', 'href': 'https://docs.readthedocs.io/en/stable/config-file/v2.html#python', 'title': '(in Read the Docs v5.20.1)', 'data-url': 'https://docs.readthedocs.io/en/stable/config-file/v2.html#python'}
_hoverxref attributes: {'class': 'reference external', 'href': 'https://docs.python.org/3/library/functions.html#float', 'title': '(in Python v3.9)', 'data-url': 'https://docs.python.org/3/library/functions.html#float'}
_hoverxref attributes: {'class': 'reference internal', 'href': '#hoverxref.extension.setup', 'title': 'hoverxref.extension.setup', 'data-project': 'myproject', 'data-version': 'myversion', 'data-doc': 'index', 'data-docpath': '/index.html', 'data-section': 'hoverxref.extension.setup'}

generating indices... genindex done
writing additional pages... search done
copying static files... done
copying extra files... done
dumping search index in English (code: en)... done
dumping object inventory... done
build succeeded.

The HTML pages are in tests/examples/intersphinx/_build/html.

# warning:

========================================================================= short test summary info ==========================================================================
FAILED tests/test_htmltag.py::test_intersphinx_default_configs - assert '<a class="reference external" href="https://docs.readthedocs.io/en/stable/config-file/v2.html#py...
FAILED tests/test_htmltag.py::test_intersphinx_python_mapping - assert '<a class="reference external" href="https://docs.readthedocs.io/en/stable/config-file/v2.html#pyt...
FAILED tests/test_htmltag.py::test_intersphinx_all_mappings - assert '<a class="hoverxref modal reference external" data-url="https://docs.readthedocs.io/en/stable/confi...
====================================================================== 3 failed, 12 passed in 12.34s =======================================================================
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-7fe5fcb0-6abe-4b94-a32d-c154f23fc1a5/test_rmtree_errorhandler_rerai0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_rmtree_errorhandler_rerai0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-7fe5fcb0-6abe-4b94-a32d-c154f23fc1a5/test_rmtree_errorhandler_reado0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_rmtree_errorhandler_reado0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-7fe5fcb0-6abe-4b94-a32d-c154f23fc1a5/test_safe_delete_no_perms0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_safe_delete_no_perms0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-7fe5fcb0-6abe-4b94-a32d-c154f23fc1a5/test_safe_set_no_perms0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_safe_set_no_perms0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-7fe5fcb0-6abe-4b94-a32d-c154f23fc1a5/test_safe_get_no_perms0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_safe_get_no_perms0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-7fe5fcb0-6abe-4b94-a32d-c154f23fc1a5
<class 'OSError'>: [Errno 39] Directory not empty: '/tmp/pytest-of-tkloczko/garbage-7fe5fcb0-6abe-4b94-a32d-c154f23fc1a5'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-5b9bd0dc-983e-4583-8b55-2bc095af2422/test_rmtree_errorhandler_rerai0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_rmtree_errorhandler_rerai0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-5b9bd0dc-983e-4583-8b55-2bc095af2422/test_rmtree_errorhandler_reado0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_rmtree_errorhandler_reado0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-5b9bd0dc-983e-4583-8b55-2bc095af2422/test_safe_delete_no_perms0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_safe_delete_no_perms0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-5b9bd0dc-983e-4583-8b55-2bc095af2422/test_safe_set_no_perms0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_safe_set_no_perms0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-5b9bd0dc-983e-4583-8b55-2bc095af2422/test_safe_get_no_perms0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_safe_get_no_perms0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-5b9bd0dc-983e-4583-8b55-2bc095af2422
<class 'OSError'>: [Errno 39] Directory not empty: '/tmp/pytest-of-tkloczko/garbage-5b9bd0dc-983e-4583-8b55-2bc095af2422'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-50c72294-1561-4b31-bd24-d8e3007c8783/test_rmtree_errorhandler_rerai0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_rmtree_errorhandler_rerai0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-50c72294-1561-4b31-bd24-d8e3007c8783/test_rmtree_errorhandler_reado0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_rmtree_errorhandler_reado0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-50c72294-1561-4b31-bd24-d8e3007c8783/test_safe_delete_no_perms0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_safe_delete_no_perms0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-50c72294-1561-4b31-bd24-d8e3007c8783/test_safe_set_no_perms0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_safe_set_no_perms0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-50c72294-1561-4b31-bd24-d8e3007c8783/test_safe_get_no_perms0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_safe_get_no_perms0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-50c72294-1561-4b31-bd24-d8e3007c8783
<class 'OSError'>: [Errno 39] Directory not empty: '/tmp/pytest-of-tkloczko/garbage-50c72294-1561-4b31-bd24-d8e3007c8783'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-5dab2e83-1e89-457e-80e6-910cfaa44e44/test_rmtree_errorhandler_rerai0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_rmtree_errorhandler_rerai0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-5dab2e83-1e89-457e-80e6-910cfaa44e44/test_rmtree_errorhandler_reado0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_rmtree_errorhandler_reado0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-5dab2e83-1e89-457e-80e6-910cfaa44e44/test_safe_delete_no_perms0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_safe_delete_no_perms0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-5dab2e83-1e89-457e-80e6-910cfaa44e44/test_safe_set_no_perms0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_safe_set_no_perms0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-5dab2e83-1e89-457e-80e6-910cfaa44e44/test_safe_get_no_perms0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_safe_get_no_perms0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-5dab2e83-1e89-457e-80e6-910cfaa44e44
<class 'OSError'>: [Errno 39] Directory not empty: '/tmp/pytest-of-tkloczko/garbage-5dab2e83-1e89-457e-80e6-910cfaa44e44'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-61f12969-bdba-42eb-92e3-8d76f137369f/test_rmtree_errorhandler_rerai0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_rmtree_errorhandler_rerai0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-61f12969-bdba-42eb-92e3-8d76f137369f/test_rmtree_errorhandler_reado0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_rmtree_errorhandler_reado0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-61f12969-bdba-42eb-92e3-8d76f137369f/test_safe_delete_no_perms0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_safe_delete_no_perms0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-61f12969-bdba-42eb-92e3-8d76f137369f/test_safe_set_no_perms0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_safe_set_no_perms0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-61f12969-bdba-42eb-92e3-8d76f137369f/test_safe_get_no_perms0
<class 'OSError'>: [Errno 39] Directory not empty: 'test_safe_get_no_perms0'
  warnings.warn(
/usr/lib/python3.8/site-packages/_pytest/pathlib.py:80: PytestWarning: (rm_rf) error removing /tmp/pytest-of-tkloczko/garbage-61f12969-bdba-42eb-92e3-8d76f137369f
<class 'OSError'>: [Errno 39] Directory not empty: '/tmp/pytest-of-tkloczko/garbage-61f12969-bdba-42eb-92e3-8d76f137369f'
  warnings.warn(
```

### PR DESCRIPTION
**Pull request by humitos**
- remove travis config
- add circleci config
- fix tests to use regex and don't match Python/Read the Docs versions (coming from intersphinx as title)

Closes #137 
Closes #106

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+jobs:
+  tests-py36:
+    docker:
+      - image: 'cimg/python:3.6'
+    steps:
+      - checkout
+      - run: pip install --user tox
+        # I didn't find a way using TOXENV with regular expression, similar to TOX_SKIP_ENV
+        # https://tox.readthedocs.io/en/latest/config.html#conf-envlist
+      - run: tox -e $(tox -l | grep -e "^py36" | paste -sd ",")
```
Comment (ericholscher) [inline .circleci/config.yml:12]: I don't really understand this comment. It should explain what this function is doing I think. Is it basically just running `tox -e py36*`

**Review by ericholscher — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (astrojuanlu) [reply .circleci/config.yml:12]: More discussion on that here https://github.com/tox-dev/tox/issues/238 (worth mentioning the issue in the comment)

Comment (humitos) [reply .circleci/config.yml:12]: > Is it basically just running `tox -e py36*`

Yes. That's what it does. However, that exact command you wrote doesn't work 😄 . So, this hack is a way to do exactly what we expect when running `tox -e py36*`

_The other_ way to achieve the same behavior is using `TOX_SKIP_ENV` with a regex that negates the matching of the envs we want.

I'm not strong on one workaround or the other. Both work. I just found the one I wrote here first.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+jobs:
+  tests-py36:
+    docker:
+      - image: 'cimg/python:3.6'
+    steps:
+      - checkout
+      - run: pip install --user tox
+        # I didn't find a way using TOXENV with regular expression, similar to TOX_SKIP_ENV
+        # https://tox.readthedocs.io/en/latest/config.html#conf-envlist
```
Comment (humitos) [inline .circleci/config.yml:11]:
```suggestion
        # Select only the environments that match the Python version we have installed 
        # in this Docker image (e.g. `py36-*`). I didn't find a way using TOXENV with
        # regular expression for this. Read more at
        # https://github.com/tox-dev/tox/issues/238 and
        # https://tox.readthedocs.io/en/latest/config.html#conf-envlist
```

@ericholscher looks better?

File: .circleci/config.yml (hunk 1 above)
Comment (humitos) [reply .circleci/config.yml]: I used `TOX_SKIP_ENV` as suggested in the issue.

Comment (stsewd) [reply .circleci/config.yml]: You can just also make this explicit like this https://github.com/readthedocs/readthedocs-sphinx-search/blob/d857859f851b210e7b2658d9efdc25c8882c0f41/.circleci/config.yml#L21

Comment (ericholscher) [reply .circleci/config.yml]: I don't care too much about the solution as long as it works. I just want to know what its doing in the comment, which worked well 👍

**Pull request merged**